cli/util/upload_bin.go

```diff
@@ -158,17 +163,16 @@ func uploadBin(ctx *cli.Context) error {
 	}
 	fmt.Fprintln(ctx.App.Writer, "First block of latest incomplete batch uploaded to NeoFS container:", oldestMissingBlockIndex)
 
+	index, partUploadedBlocks, err := uploadIndexFiles(ctx, pWrapper, containerID, acc, signer, uint(oldestMissingBlockIndex), indexFileSize, attr, indexAttrKey, homomorphicHashingDisabled, maxParallelSearches, maxRetries, debug)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("failed to upload index files: %w", err), 1)
+	}
+
 	if !ctx.Bool("skip-blocks-uploading") {
-		err = uploadBlocks(ctx, pWrapper, rpc, signer, containerID, acc, attr, oldestMissingBlockIndex, uint(currentBlockHeight), homomorphicHashingDisabled, numWorkers, maxRetries, debug)
+		err = uploadBlocks(ctx, pWrapper, rpc, signer, containerID, acc, attr, indexAttrKey, partUploadedBlocks, int(index*indexFileSize), indexFileSize, uint(currentBlockHeight), homomorphicHashingDisabled, numWorkers, maxRetries, debug)
```

Comment:
OK, I see there's a change in overall uploading logic. The updated code constructs and uploads index file once 128K batch of blocks is processed, it's OK, but: currently uploaded chains are aligned in terms of blocks/index files which means we can easily migrate uploader to the new scheme without additional call to `uploadIndexFiles`. So the question is why do we use call to `uplodIndexFiles` prior to the call `uploadBlocks`?

Also, the new scheme makes `skip-blocks-uploading` flag irrelevant, hence remove it.

Reply:
See https://github.com/nspcc-dev/neo-go/pull/3735#discussion_r1875691218.